tests/components/vesync/test_light.py

```diff
@@ -49,3 +56,75 @@ async def test_light_state(
     # Check states
     for entity in entities:
         assert hass.states.get(entity.entity_id) == snapshot(name=entity.entity_id)
+
+
+@pytest.mark.parametrize(
+    ("api_response", "expectation"),
+    [(False, pytest.raises(HomeAssistantError)), (True, NoException)],
+)
+async def test_turn_on(
+    hass: HomeAssistant,
+    light_config_entry: MockConfigEntry,
+    api_response: bool,
+    expectation,
+) -> None:
+    """Test turn_on method."""
+
+    # turn_on returns False indicating failure in which case light.turn_on
+    # raises HomeAssistantError.
+    with (
+        expectation,
+        patch(
+            "pyvesync.vesyncswitch.VeSyncDimmerSwitch.turn_on",
+            return_value=api_response,
+        ) as method_mock,
+    ):
+        with patch(
+            "homeassistant.components.vesync.light.VeSyncBaseLightHA.schedule_update_ha_state"
+        ) as update_mock:
+            await hass.services.async_call(
+                LIGHT_DOMAIN,
+                SERVICE_TURN_ON,
+                {ATTR_ENTITY_ID: ENTITY_LIGHT},
+                blocking=True,
+            )
+
+        await hass.async_block_till_done()
+        method_mock.assert_called_once()
+        update_mock.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("api_response", "expectation"),
+    [(False, pytest.raises(HomeAssistantError)), (True, NoException)],
+)
+async def test_turn_off(
+    hass: HomeAssistant,
+    light_config_entry: MockConfigEntry,
+    api_response: bool,
+    expectation,
+) -> None:
+    """Test turn_off method."""
+
+    # turn_off returns False indicating failure in which case light.turn_off
+    # raises HomeAssistantError.
+    with (
+        expectation,
+        patch(
+            "pyvesync.vesyncswitch.VeSyncDimmerSwitch.turn_off",
+            return_value=api_response,
+        ) as method_mock,
+    ):
+        with patch(
+            "homeassistant.components.vesync.light.VeSyncBaseLightHA.schedule_update_ha_state"
+        ) as update_mock:
```

Comment:
Same comment as with the switch entity

Reply:
Planning to make the change here and then update this one it is merged. https://github.com/home-assistant/core/pull/137493